### PR TITLE
Add tokens to modular tokenizer and add a large text table loader for PPI

### PR DESCRIPTION
**Pull request by floccinauc**

### CODE REVIEW

File: fusedrug/data/interaction/drug_target/loaders/indexed_text_PPI_table_loader.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+
+class IndexedTextPPITableLoader(OpBase):
+    """
+    Loads an entry from an smi file (large files are supported!).
+    """
+
+    def __init__(
+        self,
+        table_file_loc: Optional[str] = None,
```
Comment (SagiPolaczek) [inline fusedrug/data/interaction/drug_target/loaders/indexed_text_PPI_table_loader.py:14]: Why is it optional?

File: fusedrug/data/interaction/drug_target/loaders/indexed_text_PPI_table_loader.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            allow_access_by_id=self._allow_access_by_id,
+        )
+
+    def __call__(
+        self,
+        sample_dict: NDict,
+        key_out_seq: str = "data.gt",
+    ) -> NDict:
+        sid = get_sample_id(sample_dict)
+        assert isinstance(sid, int)
```
Comment (SagiPolaczek) [inline fusedrug/data/interaction/drug_target/loaders/indexed_text_PPI_table_loader.py:43]: I think that `IndexedTextTable` already take care of that (it's even a bit more permissive)

File: fusedrug/data/interaction/drug_target/loaders/indexed_text_PPI_table_loader.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+from typing import Optional
+from fuse.utils import NDict
+from fuse.data import OpBase, get_sample_id
+from fusedrug.utils.file_formats import IndexedTextTable
+
+
+class IndexedTextPPITableLoader(OpBase):
+    """
+    Loads an entry from an smi file (large files are supported!).
```
Comment (SagiPolaczek) [inline fusedrug/data/interaction/drug_target/loaders/indexed_text_PPI_table_loader.py:9]: `smi` or `txt` file?

File: fusedrug/data/interaction/drug_target/loaders/indexed_text_PPI_table_loader.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            table_file_loc,
+            seperator=self._separator,
+            id_column_idx=self._id_column_index,
+            allow_access_by_id=self._allow_access_by_id,
+        )
+
+    def __call__(
+        self,
+        sample_dict: NDict,
+        key_out_seq: str = "data.gt",
```
Comment (SagiPolaczek) [inline fusedrug/data/interaction/drug_target/loaders/indexed_text_PPI_table_loader.py:40]: not sure why `"data.gt"` should be the default value, at least not from the name of the class

**Review by SagiPolaczek — APPROVED** [review]
LGTM

I left minors inline :)

File: fusedrug/data/interaction/drug_target/loaders/indexed_text_PPI_table_loader.py (hunk 1 above)
Comment (floccinauc) [reply fusedrug/data/interaction/drug_target/loaders/indexed_text_PPI_table_loader.py:14]: remnant from someone else's code. Changed to str

File: fusedrug/data/interaction/drug_target/loaders/indexed_text_PPI_table_loader.py (hunk 3 above)
Comment (floccinauc) [reply fusedrug/data/interaction/drug_target/loaders/indexed_text_PPI_table_loader.py:9]: changed to txt

File: fusedrug/data/interaction/drug_target/loaders/indexed_text_PPI_table_loader.py (hunk 4 above)
Comment (floccinauc) [reply fusedrug/data/interaction/drug_target/loaders/indexed_text_PPI_table_loader.py:40]: changed to "data"

**Pull request merged**